File: docs/personalProjects/stats/stats.js
function normPDF(x, mu, sigma){
	return 1/Math.sqrt(2*Math.PI*sigma*sigma)*Math.pow(Math.E,-((x-mu)*(x-mu))/(2*sigma*sigma))
}

function normCDF(lower, upper, mu, sigma){
	t1 = new Date().getTime()
	let a=0 
	let step=0.000001
	for(let i=lower; i<upper; i+=step){
		a+= step*normPDF(i, mu, sigma);
	}
	t2 = new Date().getTime()
	console.log((t2-t1)/1000, " seconds")
	return a;
}

function erf(x) {
 
}

function invNorm(Area, mu, sigma){
	
}

function tPDF(x, degreesFreedom){
	
}

function tCDF(lower, upper, degreesFreedom){
	
}

function invT(Area, degreesFreedom){
	
}

File: docs/personalProjects/stats/stats.py
from math import erf, e, pi, inf, gamma
import pyerf
import pybeta

def normPDF(x, mu=0, sigma=1):
	return (1/(2*pi*(sigma**2))**.5)*(e**(-((x-mu)**2)/2*(sigma**2)))

def normCDF(low, high, mu=0, sigma=1):
	return ((1+erf((high-mu)/(sigma*2**.5)))*.5)-((1+erf((low-mu)/(sigma*2**.5)))*.5)

def invNorm(x, mu=0, sigma=1):
	return mu+sigma*((2**.5)*pyerf.erfinv(2*x-1))

def tPDF(x, df=1):
	return (gamma((df+1)/2)/((df*pi)**.5*gamma(df/2)))*((1+((x**2)/df))**(-(df+1)/2))

def tCDF(low, high, df=1):
	.5+high*gamma((df+1)/2)*(hypgeo()/((pi*df)**.5*gamma(df/2)))
	
print(tCDF())